lib/gelf/transport/tcp_tls.rb

```diff
@@ -74,16 +84,21 @@ def resource(data)
         end
       end
 
-      # These are A-level ciphers as reported from Graylog 2.0.1
-      # which were also available on Ruby using OpenSSL 1.0.2h
-      # A lot of AES-128-CBC based ciphers were not available
-      SECURE_CIPHERS = %w(
-                           AES128-GCM-SHA256
-                           ECDHE-RSA-AES128-GCM-SHA256
-                           DHE-RSA-AES128-GCM-SHA256
-                         ).freeze
+      # Ciphers have to come from the CipherString class, specifically the _TXT_ constants here - https://github.com/jruby/jruby-openssl/blob/master/src/main/java/org/jruby/ext/openssl/CipherStrings.java#L47-L178
       def restrict_ciphers(ctx)
-        ctx.ciphers = SECURE_CIPHERS
+        # This CipherString is will allow the following Ciphers
```

Comment:
I think this comment is not entirely accurate. I think the intent of this code is to only enable strong ciphers, but, two things:

1) The comment doesn't mention any more modern/stronger ciphers (AES 256 variants, etc) that are provided by default in OpenJDK and in older Oracle JRE/JDK w/ the Unlimited Strength Cryptography policy file
2) On my system, JRuby 9.1.13.0, OpenJDK 1.8.0_144, I see this list:

```
>> puts OpenSSL::SSL::SSLContext.new.tap { |c| c.ciphers = "TLSv1:!aNULL:!eNULL:!LOW:!3DES:!MD5:!EXP:!PSK:!DSS:!RC4:!SEED:!ECDSA:!ADH:!IDEA:!3DES" }.ciphers.map(&:first).sort
AES128-SHA
AES256-SHA
DHE-RSA-AES128-SHA
DHE-RSA-AES256-SHA
ECDHE-ECDSA-AES128-SHA
ECDHE-ECDSA-AES128-SHA256
ECDHE-ECDSA-AES256-SHA
ECDHE-ECDSA-AES256-SHA384
ECDHE-RSA-AES128-SHA
ECDHE-RSA-AES128-SHA256
ECDHE-RSA-AES256-SHA
ECDHE-RSA-AES256-SHA384
```

A shorter alternative to express this may be `"HIGH:!3DES"` or just `HIGH` if 3DES is acceptable.

Maybe just have the comment explain the intent "Disable any weak cipher suites" ?

Reply:
I agree, I will remove the list as it's only applicable to JRuby 1.7.
However it wasn't just strong ciphers I wanted. As the the only other option is all-ciphers, I wanted something that had ciphers that weren't cryptographically broken but according to OWASP had a broad range of compatibility.
So if you're ok with it, I will leave the cipher string as is, but add comments behind its reasoning.

Reply:
Thank you for the explanation! Sounds good to me :)